tutorials/join-across-data-sources-with-dbt.md

```diff
@@ -0,0 +1,239 @@
+And since GlareDB can connect using the Postgres protocol, you can use the
+existing Postgres adapter without needing anything new. This guide
+will walk you through setting up and using dbt with GlareDB.
+
+## Prerequisites
+
+- dbt Core installed (`pip install dbt-postgres`)
+- A GlareDB Cloud account (sign up at [console.glaredb.com])
+- Basic familiarity with dbt concepts
+
+## Setup
+
+### 1. Initialize a dbt Project
+
+Create a new dbt project:
+
+```shell
+dbt init my_glaredb_project
+```
+
+### 2. Configure dbt Credentials
+
+Add your GlareDB credentials to your `profiles.yml` file. You can find your
+connection details in GlareDB Cloud by clicking the "Connect" button and selecting
+the Postgres tab.
+
+```yaml
+dbt_glaredb_quickstart:
+  outputs:
+    dev:
+      dbname: org_name/deployment_name
+      host: proxy.glaredb.com
+      pass: your_password
+      port: 6543
+      schema: public
+      threads: 1
+      type: postgres
+      user: your_username
+  target: dev
+```
+
+> **Note**: When using your GlareDB Cloud connection string (which looks like
+> `glaredb://user_name:password@org_name.proxy.glaredb.com:6543/deployment_name`),
+> specify your `dbname` as `org_name/deployment_name` and the `host` as
+> `proxy.glaredb.com`.
+
+### 3. Test Your Connection
+
+Verify your connection:
+
+```shell
+dbt debug
+```
+
+## Creating dbt Models
+
+### Basic Model Example
+
+First, define your sources in `models/properties.yml`:
+
+```yaml
+version: 2
+
+sources:
+  - name: glaredb
+    schema: public
+    database: default
+    tables:
+      - name: sales_data
+```
+
+Then, create a staging model in `models/staging_sales.sql`:
+
+```sql
+SELECT
+    borough,
+    neighborhood,
+    building_class_category,
+    residential_units,
+    commercial_units,
+    total_units,
+    land_square_feet,
+    gross_square_feet,
+    year_built,
+    sale_price,
+    sale_date,
+    latitude,
+    longitude,
+    bin,
+    bbl
+FROM {{ source('glaredb', 'sales_data') }}
+```
+
+### Working with Multiple Data Sources
+
+GlareDB allows you to join data from different sources directly in your dbt
+models. Here's an example that combines data from an external Postgres
+data source in GlareDB and a remote file:
+
+```sql
+SELECT
+    lookup.borough_name,
+    sales.neighborhood,
+    sales.building_class_category,
+    sales.residential_units,
+    sales.commercial_units,
+    sales.total_units,
+    sales.land_square_feet,
+    sales.gross_square_feet,
+    sales.year_built,
+    sales.sale_price,
+    sales.sale_date,
+    sales.latitude,
+    sales.longitude,
+    sales.bin,
+    sales.bbl
+FROM {{ ref('staging_sales') }} sales
+JOIN my_postgres.public.borough_lookup lookup
+ON sales.borough = lookup.borough_id
+```
+
+If you won't be using your Postgres db for many queries, you may not
+want to configure it as an external data source in GlareDB. In that case, you
+could also write the previous model using the `read_postgres` function:
```

Comment:
I'd present these two examples in the inverse order so that there's some progress of "if I'm just trying things out I can use `read_postgres` and then if you want to run more queries against one postgres instance you can use the external table support.